pkg/cstor/volume/targetserver/server.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package targetserver
+
+import (
+	"encoding/json"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"syscall"
+
+	"k8s.io/klog"
+
+	cstorv1alpha1 "github.com/openebs/maya/pkg/cstor/volume/v1alpha1"
+	env "github.com/openebs/maya/pkg/env/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+var (
+	endOfData          = "\r\n"
+	respOk             = "Ok"
+	respErr            = "Err"
+	volumeMgmtUnixSock = "/var/run/volume_mgmt_sock"
+)
+
+// IsSafeToRetry will retrun true if error type is EINTR, EAGAIN or EWOULDBLOCK
+func IsSafeToRetry(err error) bool {
+	// For more information https://golang.org/pkg/syscall/#Errno
+	if err == syscall.EINTR ||
+		err == syscall.EAGAIN ||
+		err == syscall.EWOULDBLOCK {
+		return true
+	}
+	return false
+}
+
+/* Client will send below data to process
+ * Ex JSON data:
+ * {"replicaId":"6061","replicaZvolGuid":"6061",
+ * "volumeName":"vol1","replicationFactor":3,"consistencyFactor":2}
+ * and server will process it and returns Ok in case of not having error else it
+ * returns Err
+ * Response will be either "Ok" or "Err"
+ */
+
+// Reader reads the data from wire untill error or endOfData occurs
+// Reader will break only when client is sending \r\n or EOF occured
+func Reader(r io.Reader) (string, error) {
+	buf := make([]byte, 1024)
+	//collect bytes into fulllines buffer till the end of line character is reached
+	completeBytes := []byte{}
+	for {
+		n, err := r.Read(buf[:])
+		if err != nil {
+			if err == io.EOF {
+				klog.Info("Reached End Of file")
```

Comment:
'break' here makes to return error.. do we return error in such case?
And, is this not the regular case? if so, log may not be good to have here

Reply:
EOF will occur only when client closes the connection and here we are expecting \r\n as end of data as mentioned in comments. If data doesn't have \r\n better to error out it. I will remove this log.